frappe/public/js/frappe/form/layout.js

```diff
@@ -98,27 +100,40 @@ frappe.ui.form.Layout = class Layout {
 	}
 
 	show_message(html, color, permanent = false) {
-		if (this.message_color) {
-			// remove previous color
-			this.message.removeClass(this.message_color);
+		/**Render a message block with its own color and close button
+		 * @param {String} html - message or HTML to be displayed
+		 * @param {String} color - color of the block
+		 * @param {Boolean} permanent - if true, the block will not have a close button
+		 */
+		if (!html) {
```

Comment:
```suggestion
		if (html === undefined) {
```

Otherwise `frm.dashboard.set_headline("")` would clear all existing headlines, which is unexpected behavior.

Reply:
> A separate utility that handles it would be better.

We already have `frm.dashboard.clear_headline()`, which calls `set_headline` with undefined `html`. So this would still work.

>  I'm trying to keep the behaviour consistent then in this case

Makes sense. Maybe a separate PR to `develop` only, if this one is supposed to get backported.

Reply:
Yup I'll do that once this gets merged then and I'll add a todo to this PR